### DISCUSSION TITLE
Is node ordering preserved when batching multiple graphs?

### DISCUSSION THREAD

**Discussion by davidireland-iso** [pyg-team/pytorch_geometric#3038] [Q&A] [answered]
I am doing some RL so I would be interested in passing multiple graphs into a data loader and getting node embeddings for multiple graphs. I am wondering whether the node ordering is kept consistent? 

Here is my actor class 
```
class Actor(nn.Module):
    def __init__(self, input_size, hidden_size):
        super(Actor, self).__init__()
        self.input_layer = SAGEConv(input_size, hidden_size)
        self.h1 = SAGEConv(hidden_size, hidden_size)
        self.h2 = SAGEConv(hidden_size, hidden_size)
        self.output_layer = nn.Linear(hidden_size, 1)

    def forward(self, data):
        x, edge_index, batch = data.x, data.edge_index, data.batch
        x = torch.relu(self.input_layer(x, edge_index))
        x = torch.relu(self.h1(x, edge_index))
        x = torch.relu(self.h2(x, edge_index))
        probabilities = torch.sigmoid(self.output_layer(x))
        return probabilities
```

now, imagine I have passed into a data loader two graphs (batch size = 2) and the first graph has, say, 100 nodes, and the second graph has 200 nodes -- also assume WLOG that the node labels for each of the two graphs are ordered exactly from 0 to N-1, where N is total number of nodes in the graph. So graph one has node labels 0, 1, 2, ..., 99, etc -- will the first 100 rows of `probabilities` correspond to the 100 final node embeddings (in this case a scalar) of the first graph, in exact order? so the 5th row of probabilities would correspond to the node embedding of node 5 from the first graph? and e.g. the 105th row would correspond to the final node embedding of node 5 in graph 2? 

if someone could clear this up for me that'd be great, thanks 😃

**Comment by rusty1s** [comment, marked as answer]
Yes, that‘s exactly what‘s happening.

**Reply by davidireland-iso** [reply]
Thanks so much for clarifying this for me 😃
